lib/http_language_parser.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module HttpLanguageParser
+  def self.parse(header)
+    # Rails I18n uses underscores between the locale and the region; the request
+    # headers use hyphens.
+    require 'http_accept_language' unless defined? HttpAcceptLanguage
+    available_locales = I18n.available_locales.map { |locale| locale.to_s.tr('_', '-') }
+    parser = HttpAcceptLanguage::Parser.new(header)
+    parser.language_region_compatible_from(available_locales).tr('-', '_')
+  rescue
```

Comment:
Is there a specific error we're expecting here? I usually try and avoid a blank rescue since things like a typo will go unnoticed.

Reply:
I just copied/pasted the code as-is from `ApplicationController#locale_from_header`

But I do agree with you. Based on the comment, I think the rescue is there to cope with a missing header. I'll add a check for that, and remove the `rescue` :+1: